zebra-scan/src/tests.rs

```diff
@@ -166,6 +168,118 @@ async fn scanning_from_fake_generated_blocks() -> Result<()> {
     Ok(())
 }
 
+/// Scan a populated state for the ZECpages viewing key.
+/// This test is very similar to `scanning_from_populated_zebra_state` but with the ZECpages key.
+/// There are no zechub transactions in the test data so we should get empty related transactions.
+#[tokio::test]
+async fn scanning_zecpages_from_populated_zebra_state() -> Result<()> {
+    /// The extended Sapling viewing key of [ZECpages](https://zecpages.com/boardinfo)
+    const ZECPAGES_VIEWING_KEY: &str = "zxviews1q0duytgcqqqqpqre26wkl45gvwwwd706xw608hucmvfalr759ejwf7qshjf5r9aa7323zulvz6plhttp5mltqcgs9t039cx2d09mgq05ts63n8u35hyv6h9nc9ctqqtue2u7cer2mqegunuulq2luhq3ywjcz35yyljewa4mgkgjzyfwh6fr6jd0dzd44ghk0nxdv2hnv4j5nxfwv24rwdmgllhe0p8568sgqt9ckt02v2kxf5ahtql6s0ltjpkckw8gtymxtxuu9gcr0swvz";
+
+    // Parse the key from ZECpages
+    let efvk = decode_extended_full_viewing_key(
+        HRP_SAPLING_EXTENDED_FULL_VIEWING_KEY,
+        ZECPAGES_VIEWING_KEY,
+    )
+    .unwrap();
+
+    let account = AccountId::from(1);
+
+    // Build a vector of viewing keys `vks` to scan for.
+    let fvk = efvk.fvk;
+    let ivk = fvk.vk.ivk();
+    let vks: Vec<(&AccountId, &SaplingIvk)> = vec![(&account, &ivk)];
+
+    let network = zebra_chain::parameters::Network::default();
+
+    // Create a continuous chain of mainnet blocks from genesis
+    let blocks: Vec<Arc<Block>> = zebra_test::vectors::CONTINUOUS_MAINNET_BLOCKS
+        .iter()
+        .map(|(_height, block_bytes)| block_bytes.zcash_deserialize_into().unwrap())
+        .collect();
+
+    // Create a populated state service.
+    let (_state_service, read_only_state_service, latest_chain_tip, _chain_tip_change) =
+        zebra_state::populated_state(blocks.clone(), network).await;
+
+    let db = read_only_state_service.db();
+
+    // use the tip as starting height
+    let mut height = latest_chain_tip.best_tip_height().unwrap();
+
+    let mut transactions_found = 0;
+    let mut transactions_scanned = 0;
+    let mut blocks_scanned = 0;
+    while let Some(block) = db.block(height.into()) {
+        // TODO: fix this issue in zcash_client_backend
+        let sapling_tree_size = if height.is_min() {
+            1
```

Comment:
This issue never happens in `zcash_client_backend` in production, because sapling doesn't activate until block 419,200.

So I don't think a fix is needed, if you're scanning the genesis block outside tests, you're doing it wrong.

```suggestion
        // zcash_client_backend doesn't support scanning the genesis block, but that's ok, because 
        // Sapling activates at height 419,200. So we'll never scan these blocks in production code.
        let sapling_tree_size = if height.is_min() {
            1
```